services/replicator/metadataReconciler.go

```diff
@@ -335,7 +335,56 @@ func (r *metadataReconciler) reconcileCg(localCgs []*shared.ConsumerGroupDescrip
 				continue
 			}
 
-			// TODO case #2: cg exists in both remote and local, try to compare the property to see if anything gets updated
+			// case #2: cg exists in both remote and local, try to compare the property to see if anything gets updated
+			updateRequest := &shared.UpdateConsumerGroupRequest{}
+			cgUpdated := false
+
+			if localCg.GetLockTimeoutSeconds() != remoteCg.GetLockTimeoutSeconds() {
+				updateRequest.LockTimeoutSeconds = common.Int32Ptr(remoteCg.GetLockTimeoutSeconds())
+				cgUpdated = true
+			}
+			if localCg.GetMaxDeliveryCount() != remoteCg.GetMaxDeliveryCount() {
+				updateRequest.MaxDeliveryCount = common.Int32Ptr(remoteCg.GetMaxDeliveryCount())
+				cgUpdated = true
+			}
+			if localCg.GetSkipOlderMessagesSeconds() != remoteCg.GetSkipOlderMessagesSeconds() {
+				updateRequest.SkipOlderMessagesSeconds = common.Int32Ptr(remoteCg.GetSkipOlderMessagesSeconds())
+				cgUpdated = true
+			}
+			if localCg.GetStatus() != remoteCg.GetStatus() {
+				updateRequest.Status = common.InternalConsumerGroupStatusPtr(remoteCg.GetStatus())
+				cgUpdated = true
+			}
+			if localCg.GetOwnerEmail() != remoteCg.GetOwnerEmail() {
+				updateRequest.OwnerEmail = common.StringPtr(remoteCg.GetOwnerEmail())
+				cgUpdated = true
+			}
+			if localCg.GetActiveZone() != remoteCg.GetActiveZone() {
+				updateRequest.ActiveZone = common.StringPtr(remoteCg.GetActiveZone())
+				cgUpdated = true
+			}
+
+			if cgUpdated {
+				lclLg.Info(`Found cg gets updated in remote but not in local`)
```

Comment:
long line doesn't make sense to me, please revise. Also, can you emit a metric here ?